src/client/hooks/useAuth.js

```diff
@@ -0,0 +1,33 @@
+import { useEffect, useState } from 'react';
+import { auth } from '../firebase';
+
+function authRedirect() {
```

Comment:
Can we not do this in the react router?

Reply:
Not sure, when this is done in a hook, actually 🤔